contributing.rst

```diff
@@ -0,0 +1,145 @@
+
+* Please make feature requests in the same place
+
+* Please submit all patches as github pull requests
+
+Coding Guidelines
+-----------------
+
+The main rule is: *any patch that touches code must include tests.*
+And of course all tests must pass under all supported versions of Python.
+
+Fortunately that's easy to check: nose2 uses `tox`_ to manage its test
+scenarios, so simply running ``tox`` in nose2's root directory will
+run all of the tests with all supported python versions. When your
+patch gets all green, send a pull request!
+
+Some additional tips for the python and documentation in this project.
+
+- Code should be `pep8`_ compliant
+- Where possible, write code which passes ``pyflakes`` linting (consider using
+  ``flake8`` to do ``pyflakes`` and ``pep8`` checking)
+- For consistency with ``unittest`` please use CamelCase for class names,
+  methods, attributes and function parameters that map directly to class
+  attributes.
+- Try to use raw strings for docstrings -- ensures that ReST won't be
+  confused by characters like ``\\``
+- For complex functionality, include sample usage in docstrings
+- Comment liberally, but don't comment on every line of code
+- Use examples very liberally in documentation
+- Use double-quotes for strings, except when quoting a string containing
+  double-quotes but not containing single quotes
+- Use absolute imports everywhere
+- Avoid circular imports whenever possible -- given the choice between adding
+  a new module or adding a circular import, add the new module
+- Import non-``nose2`` modules and packages before importing from within
+  ``nose2``
+- Think very hard before adding a new dependency -- keep the dependencies of
+  ``nose2`` as lightweight as possible
+
+Commit Messages
+~~~~~~~~~~~~~~~
+
+A few basic ground rules for what ideal commits should look like.
+
+- No lines over 72 characters
+- No GitHub emoji -- use your words
+- Reference issues and pull requests where appropriate
+- Present tense and imperative mood
+
+Workflow and Merging
+--------------------
+
+Get Started
+~~~~~~~~~~~
+
+The ``bootstrap.sh`` script in the root of the nose2 distribution can be
+used to get a new local clone up and running quickly. It requires that
+you have `virtualenvwrapper`_ installed. Run this script once to set
+up a nose2 virtualenv, install nose2's dependencies, and set up the
+git submodule that pulls in the `Sphinx`_ theme that the docs use.
+
+Branching and Pull Requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The basic workflow should be to do the work in a topic branch in your fork
+then post a pull request for that branch.
+
+Core devs should not merge their own work -- unless it's trivial -- without
+giving other developers a chance to review it.
+
+For any pull request,
+
+- *Make sure it meets the standards set in this document*
+- *Make sure it merges cleanly*
+- *List any issues closed by the pull request*
+- *Squash intermediate commits*. Use ``git rebase --interactive`` to squash
```

Comment:
Different maintainers have different views on this. My view is make commit history pretty, then rebase and merge the PR. Other folks have the view of never force-push, squash and merge at the end (or create a merge commit as below). 

I don't have a hugely strong preference about merging.

Reply:
Yes, this is subject to many different opinions. Would you prefer that the contrib guide not express any guidance on this front?

With or without guidance, there will inevitably be pull requests with commit history like this:
```
shaxyz1 "Fix foo"
shaxyz2 "Revert 'Fix foo'"
shaxyz3 "Fix foo (but really)"
shaxyz4 "Typo"
shaxyz5 "Fix bar too"
shaxyz6 "Typo"
```
which should really just be
```
shaxyz7 "Fix foo"
shaxyz8 "Fix bar too"
```

It's unlikely that a line in the contributing doc will have any impact, so it's really just a matter of having something authoritative stating policy if you want to make people cleanup such PRs.

Reply:
What you put is good.